pkg/services/folder/folderimpl/folder.go

```diff
@@ -284,6 +287,93 @@ func (s *Service) DeleteFolder(ctx context.Context, user *user.SignedInUser, org
 	return dashFolder, nil
 }
 
+func (s *Service) Create(ctx context.Context, cmd *folder.CreateFolderCommand) (*folder.Folder, error) {
+	// check the flag, if old - do whatever did before
+	//  for new only the store
+	cmd.UID = util.GenerateShortUID()
+	return s.store.Create(ctx, *cmd)
+}
+
+func (s *Service) Update(ctx context.Context, cmd *folder.UpdateFolderCommand) (*folder.Folder, error) {
+	// check the flag, if old - do whatever did before
+	//  for new only the store
+	return s.store.Update(ctx, *cmd)
+}
+
+func (s *Service) Move(ctx context.Context, cmd *folder.MoveFolderCommand) (*folder.Folder, error) {
+	// check the flag, if old - do whatever did before
+	//  for new only the store
+	//  use Update
+	return nil, s.store.Move(ctx, *cmd)
```

Comment:
We must ensure there are no loops, either here in the service, or at the store layer.
I.e. we should get parents of a destination and make sure that current folder UID is not in that list.
I.e. for folder hierarcy `/A/B/C/D/E/F` moving `B` into `E` or `F` or `B` itself should not be possible.
We can implement this check in a separate PR, but at least a TODO comment would be nice to have.